announcement/announcement-tool/tool/src/java/org/sakaiproject/announcement/tool/AnnouncementAction.java

```diff
@@ -2863,8 +2864,30 @@ else if (announceTo != null && announceTo.equals("groups"))
 					{
 						// availablity changed
 						eventTrackingService.post(eventTrackingService.newEvent(AnnouncementService.EVENT_ANNC_UPDATE_AVAILABILITY, msg.getReference(), true));
+
+						//SAK-44622:
+						//check if an delay might exist
+						if ((StringUtils.isNotEmpty(oReleaseDate) && releaseDate == null) || (StringUtils.isNotEmpty(oReleaseDate) && releaseDate != null && !oReleaseDate.equals(releaseDate.toString()))) {
+
+							eventTrackingService.cancelDelays(msg.getReference(), org.sakaiproject.announcement.api.AnnouncementService.EVENT_AVAILABLE_ANNC);
+
+							//check if new date is in future
```

Comment:
Shouldn't this be "check if new date has passed already" ? Or something like that?

Reply:
done